---
title: "Single cell transcriptomes of macrophages with HIV 2024 - data preparation"
author: "Burnet Bioinformatics group"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    fig_width: 7
    fig_height: 7
theme: cosmo
---

## Introduction

Description of the samples TODO.

```{r,libs}

suppressPackageStartupMessages({
  library("plyr")
  library("Seurat")
  library("hdf5r")
  library("SingleCellExperiment")
  library("parallel")
  library("stringi")
  library("beeswarm")
  library("muscat")
  library("DESeq2")
  library("mitch")
  library("vioplot")
  library("beeswarm")
})

```

## Load data

Load the h5 matrices.

CSP1_counts ./1-CSP/outs/filtered_feature_bc_matrix.h5

GEX1_counts ./1-GEX/outs/filtered_feature_bc_matrix.h5

CSP2_counts ./2-CSP/outs/filtered_feature_bc_matrix.h5

GEX2_counts ./2-GEX/outs/filtered_feature_bc_matrix.h5

M239_N239_Alv_counts ./M239-N239-Alv/outs/filtered_feature_bc_matrix.h5

N239Alv_react ./N239Alv-react/outs/filtered_feature_bc_matrix.h5

P239_Alv_and_MDM ./P239-Alv-and-MDM/outs/filtered_feature_bc_matrix.h5

```{r,load1}

CSP1_counts <- Read10X_h5("./1-CSP/outs/filtered_feature_bc_matrix.h5")
colnames(CSP1_counts) <- gsub("-1","",colnames(CSP1_counts))
dim(CSP1_counts)
ncol(CSP1_counts)
summary(colSums(CSP1_counts))
CSP1_counts[1:10,1:30]

GEX1_counts <- Read10X_h5("./1-GEX/outs/filtered_feature_bc_matrix.h5")
colnames(GEX1_counts) <- gsub("-1","",colnames(GEX1_counts))
ncol(GEX1_counts)
summary(colSums(GEX1_counts))
GEX1_counts[1:10,1:30]

CSP2_counts <- Read10X_h5("./2-CSP/outs/filtered_feature_bc_matrix.h5")
colnames(CSP2_counts) <- gsub("-1","",colnames(CSP2_counts))
ncol(CSP2_counts)
summary(colSums(CSP2_counts))
CSP2_counts[1:10,1:30]

GEX2_counts <- Read10X_h5("./2-GEX/outs/filtered_feature_bc_matrix.h5")
colnames(GEX2_counts) <- gsub("-1","",colnames(GEX2_counts))
ncol(GEX2_counts)
summary(colSums(GEX2_counts))
GEX2_counts[1:10,1:30]

M239_N239_Alv_counts <- Read10X_h5("./M239-N239-Alv/outs/filtered_feature_bc_matrix.h5")
colnames(M239_N239_Alv_counts) <- gsub("-1","",colnames(M239_N239_Alv_counts))
ncol(M239_N239_Alv_counts)
summary(colSums(M239_N239_Alv_counts))
M239_N239_Alv_counts[1:10,1:30]

N239Alv_react_counts <- Read10X_h5("./N239Alv-react/outs/filtered_feature_bc_matrix.h5")
colnames(N239Alv_react_counts) <- gsub("-1","",colnames(N239Alv_react_counts))
ncol(N239Alv_react_counts)
summary(colSums(N239Alv_react_counts))
N239Alv_react_counts[1:10,1:30]

P239_Alv_MDM_counts <- Read10X_h5("./P239-Alv-and-MDM/outs/filtered_feature_bc_matrix.h5")
colnames(P239_Alv_MDM_counts) <- gsub("-1","",colnames(P239_Alv_MDM_counts))
ncol(P239_Alv_MDM_counts)
summary(colSums(P239_Alv_MDM_counts))
P239_Alv_MDM_counts[1:10,1:30]

```

CSP1_counts ./1-CSP/outs/filtered_feature_bc_matrix.h5

GEX1_counts ./1-GEX/outs/filtered_feature_bc_matrix.h5

CSP2_counts ./2-CSP/outs/filtered_feature_bc_matrix.h5

GEX2_counts ./2-GEX/outs/filtered_feature_bc_matrix.h5

M239_N239_Alv_counts ./M239-N239-Alv/outs/filtered_feature_bc_matrix.h5

N239Alv_react_counts ./N239Alv-react/outs/filtered_feature_bc_matrix.h5

P239_Alv_MDM_counts ./P239-Alv-and-MDM/outs/filtered_feature_bc_matrix.h5

```{r,cellbarcodes}

if (! file.exists("CSP1_cell_barcodes.txt") ) {
  CSP1_bc <- colnames(CSP1_counts)
  writeLines(CSP1_bc,con="CSP1_cell_barcodes.txt")
}

if (! file.exists("GEX1_cell_barcodes.txt") ) {
  GEX1_bc <- colnames(GEX1_counts)
  writeLines(GEX1_bc,con="GEX1_cell_barcodes.txt")
}

if (! file.exists("CSP2_cell_barcodes.txt") ) {
  CSP2_bc <- colnames(CSP2_counts)
  writeLines(CSP2_bc,con="CSP2_cell_barcodes.txt")
}

if (! file.exists("GEX2_cell_barcodes.txt") ) {
  GEX2_bc <- colnames(GEX2_counts)
  writeLines(GEX2_bc,con="GEX2_cell_barcodes.txt")
}

if (! file.exists("M239_N239_Alv_cell_barcodes.txt") ) {
  M239_N239_Alv_bc <- colnames(M239_N239_Alv_counts)
  writeLines(M239_N239_Alv_bc,con="M239_N239_Alv_cell_barcodes.txt")
}

if (! file.exists("N239Alv_react_cell_barcodes.txt") ) {
  N239Alv_react_bc <- colnames(N239Alv_react_counts)
  writeLines(N239Alv_react_bc,con="N239Alv_react_cell_barcodes.txt")
}

if (! file.exists("P239_Alv_and_MDM_cell_barcodes.txt") ) {
  P239_Alv_and_MDM_bc <- colnames(P239_Alv_MDM_counts)
  writeLines(P239_Alv_and_MDM_bc,con="P239_Alv_and_MDM_cell_barcodes.txt")
}

```

Now read HTO data.

```{r,htoread1}

gex1_hto <- Read10X("1-CSP_hto/read_count/", gene.column=1)
#gex1_hto_trash <- Read10X("trash/1-CSP_hto/read_count/", gene.column=1)
summary(colSums(gex1_hto))
dim(gex1_hto)

gex2_hto <- Read10X("2-CSP_hto/read_count/", gene.column=1)
summary(colSums(gex2_hto))
dim(gex2_hto)

M239_N239_Alv_hto <- Read10X("M239-N239-Alv_hto/read_count/", gene.column=1)
summary(colSums(M239_N239_Alv_hto))
dim(M239_N239_Alv_hto)

P239_Alv_MDM_hto <- Read10X("P239-Alv-and-MDM_hto/read_count/", gene.column=1)
summary(colSums(P239_Alv_MDM_hto))
dim(P239_Alv_MDM_hto)

str(which(colnames(GEX1_counts) %in% colnames(gex1_hto)))
str(which(colnames(GEX2_counts) %in% colnames(gex2_hto)))
str(which(colnames(M239_N239_Alv_counts) %in% colnames(M239_N239_Alv_hto)))
str(which(colnames(P239_Alv_MDM_counts) %in% colnames(P239_Alv_MDM_hto)))

```

## Remove low HTO counts

```{r,htofilt1}

summary(colSums(gex1_hto))
table(colSums(gex1_hto)>=100)
gex1_hto <- gex1_hto[,which(colSums(gex1_hto)>=100)]
summary(colSums(gex1_hto))

summary(colSums(gex2_hto))
table(colSums(gex2_hto)>=100)
gex2_hto <- gex2_hto[,which(colSums(gex2_hto)>=100)]
summary(colSums(gex2_hto))

summary(colSums(M239_N239_Alv_hto))
table(colSums(M239_N239_Alv_hto)>=100)
M239_N239_Alv_hto <- M239_N239_Alv_hto[,which(colSums(M239_N239_Alv_hto)>=100)]
summary(colSums(M239_N239_Alv_hto))

summary(colSums(P239_Alv_MDM_hto))
table(colSums(P239_Alv_MDM_hto)>=100)
P239_Alv_MDM_hto <- P239_Alv_MDM_hto[,which(colSums(P239_Alv_MDM_hto)>=100)]
summary(colSums(P239_Alv_MDM_hto))

```

## Look at the proportion unmapped

```{r,htofilt2}

summary(apply(gex1_hto,2,function(x) {x[6]/sum(x) } ) )
summary(apply(gex2_hto,2,function(x) {x[6]/sum(x) } ) )
summary(apply(M239_N239_Alv_hto,2,function(x) {x[6]/sum(x) } ) )
summary(apply(P239_Alv_MDM_hto,2,function(x) {x[6]/sum(x) } ) )

```

## Identify ambiguous HTO counts

For each cell barcode, calculate the ratio of top BC to 2nd BC.

```{r,htodoublets1}

getratio <- function(mx){
  res <- lapply(1:ncol(mx), function(i) {
    cnt <- mx[,i]
    top1 <- cnt[order(-cnt)][1]+1
    top2 <- cnt[order(-cnt)][2]+1
    top1/top2
  })
  return(unlist(res))
}

gex1_hto_ratio <- getratio(gex1_hto)
summary(unlist(gex1_hto_ratio))
table(gex1_hto_ratio>3)
gex1_hto <- gex1_hto[,which(gex1_hto_ratio>3)]

gex2_hto_ratio <- getratio(gex2_hto)
summary(unlist(gex2_hto_ratio))
table(gex2_hto_ratio>3)
gex2_hto <- gex2_hto[,which(gex2_hto_ratio>3)]

M239_N239_Alv_hto_ratio <- getratio(M239_N239_Alv_hto)
summary(unlist(M239_N239_Alv_hto_ratio))
table(M239_N239_Alv_hto_ratio>3)
M239_N239_Alv_hto <- M239_N239_Alv_hto[,which(M239_N239_Alv_hto_ratio>3)]

P239_Alv_MDM_hto_ratio <- getratio(P239_Alv_MDM_hto)
summary(unlist(P239_Alv_MDM_hto_ratio))
table(P239_Alv_MDM_hto_ratio>3)
P239_Alv_MDM_hto <- P239_Alv_MDM_hto[,which(P239_Alv_MDM_hto_ratio>3)]

```

## Match 

```{r,match1}

gex1_itx <- intersect(colnames(GEX1_counts),colnames(gex1_hto))
dim(GEX1_counts)
GEX1_counts <- GEX1_counts[,gex1_itx]
dim(GEX1_counts)
gex1_hto <- gex1_hto[,gex1_itx]
dim(gex1_hto)

gex2_itx <- intersect(colnames(GEX2_counts),colnames(gex2_hto))
dim(GEX2_counts)
GEX2_counts <- GEX2_counts[,gex2_itx]
dim(GEX2_counts)
gex2_hto <- gex2_hto[,gex2_itx]
dim(gex2_hto)

M239_N239_Alv_itx <- intersect(colnames(M239_N239_Alv_counts),colnames(M239_N239_Alv_hto))
dim(M239_N239_Alv_counts)
M239_N239_Alv_counts <- M239_N239_Alv_counts[,M239_N239_Alv_itx]
dim(M239_N239_Alv_counts)
M239_N239_Alv_hto <- M239_N239_Alv_hto[,M239_N239_Alv_itx]
dim(M239_N239_Alv_hto)

P239_Alv_MDM_itx <- intersect(colnames(P239_Alv_MDM_counts),colnames(P239_Alv_MDM_hto))
dim(P239_Alv_MDM_counts)
P239_Alv_MDM_counts <- P239_Alv_MDM_counts[,P239_Alv_MDM_itx]
dim(P239_Alv_MDM_counts)
P239_Alv_MDM_hto <- P239_Alv_MDM_hto[,P239_Alv_MDM_itx]
dim(P239_Alv_MDM_hto)

```

## Get demux result

Hash tagging sheet.

| Library | HTO | Sample |
| --- | --- | --- |
| gex1 | 1 | mock (donor P and V) |
| gex1 | 2 | gfp_neg (donor P and V) |
| gex1 | 4 | gfp_pos (donor P and V) |
| gex2 | 1 | mock donor O |
| gex2 | 2 | gfp_neg donor O |
| gex2 | 4 | gfp_pos donor O |
| M239_N239_Alv | 1 | AlvM239_mock |
| M239_N239_Alv | 2 | AlvM239_gfp_neg |
| M239_N239_Alv | 3 | AlvM239_gfp_pos |
| M239_N239_Alv | 4 | AlvN239_mock |
| M239_N239_Alv | 5 | AlvN239_gfp_neg |
| M239_N239_Alv | 6 | AlvN239_gfp_pos |
| P239_Alv_MDM_hto | 1 | MDM_mock |
| P239_Alv_MDM_hto | 2 | MDM_gfp_neg |
| P239_Alv_MDM_hto | 3 | MDM_gfp_pos |
| P239_Alv_MDM_hto | 4 | Alv_mock |
| P239_Alv_MDM_hto | 5 | Alv_gfp_neg |
| P239_Alv_MDM_hto | 6 | Alv_gfp_pos |


Input counts:

* GEX1_counts gex1_hto

* GEX2_counts gex2_hto

* M239_N239_Alv M239_N239_Alv_hto

* P239_Alv_MDM P239_Alv_MDM_hto

```{r,demuxresult1}

table(apply(gex1_hto,2,function(x) { order(-x) } )[1,] )
gex1_idx <- apply(gex1_hto,2,function(x) { order(-x) } )[1,]
gex1_h1 <- GEX1_counts[,which(gex1_idx==1)] # mock (donor P and V)
gex1_h2 <- GEX1_counts[,which(gex1_idx==2)] # gfp_neg (donor P and V) 
gex1_h4 <- GEX1_counts[,which(gex1_idx==4)] # gfp_pos (donor P and V)

table(apply(gex2_hto,2,function(x) { order(-x) } )[1,] )
gex2_idx <- apply(gex2_hto,2,function(x) { order(-x) } )[1,]
gex2_h1 <- GEX2_counts[,which(gex2_idx==1)] # mock donor O
gex2_h2 <- GEX2_counts[,which(gex2_idx==2)] # gfp_neg donor O 
gex2_h4 <- GEX2_counts[,which(gex2_idx==4)] # gfp_pos donor O

table(apply(M239_N239_Alv_hto,2,function(x) { order(-x) } )[1,] )
M239_N239_Alv_idx <- apply(M239_N239_Alv_hto,2,function(x) { order(-x) } )[1,]
M239_N239_Alv_h1 <- M239_N239_Alv_counts[,which(M239_N239_Alv_idx==1)] # AlvM239_mock
M239_N239_Alv_h2 <- M239_N239_Alv_counts[,which(M239_N239_Alv_idx==2)] # AlvM239_gfp_neg
M239_N239_Alv_h3 <- M239_N239_Alv_counts[,which(M239_N239_Alv_idx==3)] # AlvM239_gfp_pos
M239_N239_Alv_h4 <- M239_N239_Alv_counts[,which(M239_N239_Alv_idx==4)] # AlvN239_mock
M239_N239_Alv_h5 <- M239_N239_Alv_counts[,which(M239_N239_Alv_idx==5)] # AlvN239_gfp_neg        
M239_N239_Alv_h6 <- M239_N239_Alv_counts[,which(M239_N239_Alv_idx==6)] # AlvN239_gfp_pos     

table(apply(P239_Alv_MDM_hto,2,function(x) { order(-x) } )[1,] )
P239_Alv_MDM_idx <- apply(P239_Alv_MDM_hto,2,function(x) { order(-x) } )[1,]
P239_Alv_MDM_h1 <- P239_Alv_MDM_counts[,which(P239_Alv_MDM_idx==1)] # MDM_mock
P239_Alv_MDM_h2 <- P239_Alv_MDM_counts[,which(P239_Alv_MDM_idx==2)] # MDM_gfp_neg
P239_Alv_MDM_h3 <- P239_Alv_MDM_counts[,which(P239_Alv_MDM_idx==3)] # MDM_gfp_pos
P239_Alv_MDM_h4 <- P239_Alv_MDM_counts[,which(P239_Alv_MDM_idx==4)] # Alv_mock
P239_Alv_MDM_h5 <- P239_Alv_MDM_counts[,which(P239_Alv_MDM_idx==5)] # Alv_gfp_neg
P239_Alv_MDM_h6 <- P239_Alv_MDM_counts[,which(P239_Alv_MDM_idx==6)] # Alv_gfp_pos

```

## Distinguish SNP pooled samples

Data was previously genotyped and called using cellSNP-lite and vireo.

```{r,vireo}

snp <- read.table("1-GEX_vireo/donor_ids.tsv",header=TRUE)

donor0 <- subset(snp,donor_id=="donor0")$cell
donor0 <- gsub("-1","",donor0)

donor1 <- subset(snp,donor_id=="donor1")$cell
donor1 <- gsub("-1","",donor1)

gex1_h1_d0 <- gex1_h1[,colnames(gex1_h1) %in% donor0]
dim(gex1_h1_d0)

gex1_h1_d1 <- gex1_h1[,colnames(gex1_h1) %in% donor1]
dim(gex1_h1_d1)

gex1_h2_d0 <- gex1_h2[,colnames(gex1_h2) %in% donor0]
dim(gex1_h2_d0)

gex1_h2_d1 <- gex1_h2[,colnames(gex1_h2) %in% donor1]
dim(gex1_h2_d1)

gex1_h4_d0 <- gex1_h4[,colnames(gex1_h4) %in% donor0]
dim(gex1_h4_d0)

gex1_h4_d1 <- gex1_h4[,colnames(gex1_h4) %in% donor1]
dim(gex1_h4_d1)

```

## Which is female or male?

Let's look at the expression of genes on sex chromosomes.

BTW here's the bash script to get the chr2gene tsv file:

```
zcat genes.gtf.gz \
| grep ^chr \
| cut -f1,9 \
| sed 's/; /\n/g' \
| egrep  '(chr|gene_name)' \
| sed 's/gene_name "//' \
| tr -d '"' \
| cut -f1 \
| paste - - \
| uniq \
| sort -u > gene_chr.tsv
```

```{r,chromosomes1}

gene_chr <- read.table("ref_combined1/hum_hiv/genes/gene_chr.tsv")
head(gene_chr)

chrx <- subset(gene_chr,V1=="chrX")$V2
chry <- subset(gene_chr,V1=="chrY")$V2
chrm <- subset(gene_chr,V1=="chrM")$V2
chra <- gene_chr$V2[!gene_chr$V2 %in% c(chrx,chry,chrm)]
str(gene_chr)
str(chrx)
str(chry)
str(chrm)

d0y <- median(colSums(gex1_h4_d0[which(rownames(gex1_h4_d0) %in% chry),]))
d0x <- median(colSums(gex1_h4_d0[which(rownames(gex1_h4_d0) %in% chrx),]))
d0m <- median(colSums(gex1_h4_d0[which(rownames(gex1_h4_d0) %in% chrm),]))
d0a <- median(colSums(gex1_h4_d0[which(rownames(gex1_h4_d0) %in% chra),]))

d1y <- median(colSums(gex1_h4_d1[which(rownames(gex1_h4_d1) %in% chry),]))
d1x <- median(colSums(gex1_h4_d1[which(rownames(gex1_h4_d1) %in% chrx),]))
d1m <- median(colSums(gex1_h4_d1[which(rownames(gex1_h4_d1) %in% chrm),]))
d1a <- median(colSums(gex1_h4_d1[which(rownames(gex1_h4_d1) %in% chra),]))

message(paste("Donor 0 chrY:",signif(d0y/d0a * 100,3),"%"))
message(paste("Donor 0 chrX:",signif(d0x/d0a * 100,3),"%"))
message(paste("Donor 0 chrM:",signif(d0m/d0a * 100,3),"%"))

message(paste("Donor 1 chrY:",signif(d1y/d1a * 100,3),"%"))
message(paste("Donor 1 chrX:",signif(d1x/d1a * 100,3),"%"))
message(paste("Donor 1 chrM:",signif(d1m/d1a * 100,3),"%"))

```

Therefore donor 0 is female and 1 is male.

## Check HIV in these

HIV genes are rows 1 to 21.

```{r,checkhiv}

hiv <- mean(colSums(gex1_h1_d0[1:21,])) / mean(colSums(gex1_h1_d0[22:nrow(gex1_h1_d0),]))
message(paste("Mock Donor 0 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex1_h1_d1[1:21,])) / mean(colSums(gex1_h1_d1[22:nrow(gex1_h1_d1),]))
message(paste("Mock Donor 1 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex1_h2_d0[1:21,])) / mean(colSums(gex1_h2_d0[22:nrow(gex1_h2_d0),]))
message(paste("GFP- Donor 0 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex1_h2_d1[1:21,])) / mean(colSums(gex1_h2_d1[22:nrow(gex1_h2_d1),]))
message(paste("GFP- Donor 1 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex1_h4_d0[1:21,])) / mean(colSums(gex1_h4_d0[22:nrow(gex1_h4_d0),]))
message(paste("GFP+ Donor 0 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex1_h4_d1[1:21,])) / mean(colSums(gex1_h4_d1[22:nrow(gex1_h4_d1),]))
message(paste("GFP+ Donor 1 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex2_h1[1:21,])) / mean(colSums(gex2_h1[22:nrow(gex2_h1),]))
message(paste("Mock Donor 2 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex2_h2[1:21,])) / mean(colSums(gex2_h2[22:nrow(gex2_h2),]))
message(paste("GFP- Donor 2 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(gex2_h4[1:21,])) / mean(colSums(gex2_h4[22:nrow(gex2_h4),]))
message(paste("GFP+ Donor 2 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(P239_Alv_MDM_h1[1:21,])) / mean(colSums(P239_Alv_MDM_h1[22:nrow(P239_Alv_MDM_h1),]))
message(paste("Mock Donor 3 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(P239_Alv_MDM_h2[1:21,])) / mean(colSums(P239_Alv_MDM_h2[22:nrow(P239_Alv_MDM_h2),]))
message(paste("GFP+ Donor 3 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(P239_Alv_MDM_h3[1:21,])) / mean(colSums(P239_Alv_MDM_h3[22:nrow(P239_Alv_MDM_h3),]))
message(paste("GPF- Donor 3 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(P239_Alv_MDM_h4[1:21,])) / mean(colSums(P239_Alv_MDM_h4[22:nrow(P239_Alv_MDM_h4),]))
message(paste("Mock Donor 4 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(P239_Alv_MDM_h5[1:21,])) / mean(colSums(P239_Alv_MDM_h5[22:nrow(P239_Alv_MDM_h5),]))
message(paste("GFP+ Donor 4 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(P239_Alv_MDM_h6[1:21,])) / mean(colSums(P239_Alv_MDM_h6[22:nrow(P239_Alv_MDM_h6),]))
message(paste("GFP- Donor 4 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(M239_N239_Alv_h1[1:21,])) / mean(colSums(M239_N239_Alv_h1[22:nrow(M239_N239_Alv_h1),]))
message(paste("Mock Donor 5 HIV:",signif(hiv,3),"%"))

hiv <- mean(colSums(M239_N239_Alv_h2[1:21,])) / mean(colSums(M239_N239_Alv_h2[22:nrow(M239_N239_Alv_h2),]))
message(paste("Mock Donor 5 HIV:",signif(hiv,3),"%"))

```

Yes it looks like GFP minus and positive were swapped

## Renaming things

From now on we will rename them: "mock", "latent", "active"

mock0 -> mdm_mock1
mock1 -> mdm_mock2
mock2 -> mdm_mock3
mock3 -> mdm_mock4

mock4 -> alv_mock1
mock5 -> alv_mock3
mock6 -> alv_mock4

active0 -> mdm_active1
active1 -> mdm_active2
active2 -> mdm_active3
active3 -> mdm_active4

active4 -> alv_active1
active5 -> alv_active2
active6 -> alv_active3

latent0 -> mdm_latent1
latent1 -> mdm_latent2
latent2 -> mdm_latent3
latent3 -> mdm_latent4

latent4 -> alv_latent1
latent5 -> alv_latent2
latent6 -> alv_latent3

bystander0 -> mdm_bystander1
bystander1 -> mdm_bystander2
bystander2 -> mdm_bystander3
bystander3 -> mdm_bystander4

bystander4 -> alv_bystander1
bystander5 -> alv_bystander2
bystander6 -> alv_bystander3

```{r,rename1,fig.height=9}

mdm_mock1 <- gex1_h1_d0 #MDMP236 MDMmock
mdm_mock2 <- gex1_h1_d1 #MDMV236 MDMmock
mdm_mock3 <- gex2_h1 #MDMO236 MDMmock
mdm_mock4 <- P239_Alv_MDM_h1 #MDMP239 MDMmock

alv_mock1 <- P239_Alv_MDM_h4 #AlvP239 AlvMDMmock
alv_mock2 <- M239_N239_Alv_h1 #AlvM239 AlvMDMmock
alv_mock3 <- M239_N239_Alv_h4 #AlvN239 AlvMDMmock

mdm_active1 <- gex1_h2_d0 #MDMP236 MDMactive
mdm_active2 <- gex1_h2_d1 #MDMV236 MDMactive
mdm_active3 <- gex2_h2 #MDMO236 MDMactive
mdm_active4 <- P239_Alv_MDM_h2 #MDMP239 MDMactive

alv_active1 <- P239_Alv_MDM_h5 #AlvP239 AlvMDMactive
alv_active2 <- M239_N239_Alv_h2 #AlvM239 AlvMDMactive
alv_active3 <- M239_N239_Alv_h5 #AlvN239 AlvMDMactive

mdm_gfpneg1 <- colSums(gex1_h4_d0[1:21,])/colSums(gex1_h4_d0)*1e6
mdm_gfpneg2 <- colSums(gex1_h4_d1[1:21,])/colSums(gex1_h4_d1)*1e6
mdm_gfpneg3 <- colSums(gex2_h4[1:21,])/colSums(gex2_h4)*1e6
mdm_gfpneg4 <- colSums(P239_Alv_MDM_h3[1:21,])/colSums(P239_Alv_MDM_h3)*1e6

alv_gfpneg1 <- colSums(P239_Alv_MDM_h6[1:21,])/colSums(P239_Alv_MDM_h6)*1e6
alv_gfpneg2 <- colSums(M239_N239_Alv_h3[1:21,])/colSums(M239_N239_Alv_h3)*1e6
alv_gfpneg3 <- colSums(M239_N239_Alv_h6[1:21,])/colSums(M239_N239_Alv_h6)*1e6

gfp <- list("mdm_mock1"=colSums(mdm_mock1[1:21,])/colSums(mdm_mock1)*1e6 ,
  "mdm_mock2"=colSums(mdm_mock2[1:21,])/colSums(mdm_mock2)*1e6 ,
  "mdm_mock3"=colSums(mdm_mock3[1:21,])/colSums(mdm_mock3)*1e6 ,
  "mdm_mock4"=colSums(mdm_mock4[1:21,])/colSums(mdm_mock4)*1e6 ,
  "alv_mock1"=colSums(alv_mock1[1:21,])/colSums(alv_mock1)*1e6 ,
  "alv_mock2"=colSums(alv_mock2[1:21,])/colSums(alv_mock2)*1e6 ,
  "alv_mock3"=colSums(alv_mock3[1:21,])/colSums(alv_mock3)*1e6 ,
  "mdm_active1"=colSums(mdm_active1[1:21,])/colSums(mdm_active1)*1e6,
  "mdm_active2"=colSums(mdm_active2[1:21,])/colSums(mdm_active2)*1e6,
  "mdm_active3"=colSums(mdm_active3[1:21,])/colSums(mdm_active3)*1e6,
  "mdm_active4"=colSums(mdm_active4[1:21,])/colSums(mdm_active4)*1e6,
  "alv_active1"=colSums(alv_active1[1:21,])/colSums(alv_active1)*1e6,
  "alv_active2"=colSums(alv_active2[1:21,])/colSums(alv_active2)*1e6,
  "alv_active3"=colSums(alv_active3[1:21,])/colSums(alv_active3)*1e6,
  "mdm_gfpneg1"=colSums(gex1_h4_d0[1:21,])/colSums(gex1_h4_d0)*1e6,
  "mdm_gfpneg2"=colSums(gex1_h4_d1[1:21,])/colSums(gex1_h4_d1)*1e6,
  "mdm_gfpneg3"=colSums(gex2_h4[1:21,])/colSums(gex2_h4)*1e6,
  "mdm_gfpneg4"=colSums(P239_Alv_MDM_h3[1:21,])/colSums(P239_Alv_MDM_h3)*1e6,
  "alv_gfpneg1"=colSums(P239_Alv_MDM_h6[1:21,])/colSums(P239_Alv_MDM_h6)*1e6,
  "alv_gfpneg2"=colSums(M239_N239_Alv_h3[1:21,])/colSums(M239_N239_Alv_h3)*1e6,
  "alv_gfpneg3"=colSums(M239_N239_Alv_h6[1:21,])/colSums(M239_N239_Alv_h6)*1e6)

par(mar=c(7.1, 4.1, 4.1, 2.1))

lgfp <- lapply(gfp,function(x) { log10(x+1) } )
boxplot(lgfp,col="white",cex=0,las=2,border="darkgray",ylab="log10+1 RPM",main="All HIV genes")
boxplot(lgfp,col="white",cex=0,las=2,border="darkgray",ylab="log10+1 RPM",main="All HIV genes")
beeswarm(lgfp,add=TRUE,pch=1,cex=0.1)

nullres <- lapply(1:21, function(i) {

  gfp <- list("mdm_mock1"=mdm_mock1[i,]/colSums(mdm_mock1)*1e6 ,
    "mdm_mock2"=mdm_mock2[i,]/colSums(mdm_mock2)*1e6 ,
    "mdm_mock3"=mdm_mock3[i,]/colSums(mdm_mock3)*1e6 ,
    "mdm_mock4"=mdm_mock4[i,]/colSums(mdm_mock4)*1e6 ,
    "alv_mock1"=alv_mock1[i,]/colSums(alv_mock1)*1e6 ,
    "alv_mock2"=alv_mock2[i,]/colSums(alv_mock2)*1e6 ,
    "alv_mock3"=alv_mock3[i,]/colSums(alv_mock3)*1e6 ,
    "mdm_active1"=mdm_active1[i,]/colSums(mdm_active1)*1e6,
    "mdm_active2"=mdm_active2[i,]/colSums(mdm_active2)*1e6,
    "mdm_active3"=mdm_active3[i,]/colSums(mdm_active3)*1e6,
    "mdm_active4"=mdm_active4[i,]/colSums(mdm_active4)*1e6,
    "alv_active1"=alv_active1[i,]/colSums(alv_active1)*1e6,
    "alv_active2"=alv_active2[i,]/colSums(alv_active2)*1e6,
    "alv_active3"=alv_active3[i,]/colSums(alv_active3)*1e6,
    "mdm_gfpneg1"=gex1_h4_d0[i,]/colSums(gex1_h4_d0)*1e6,
    "mdm_gfpneg2"=gex1_h4_d1[i,]/colSums(gex1_h4_d1)*1e6,
    "mdm_gfpneg3"=gex2_h4[i,]/colSums(gex2_h4)*1e6,
    "mdm_gfpneg4"=P239_Alv_MDM_h3[i,]/colSums(P239_Alv_MDM_h3)*1e6,
    "alv_gfpneg1"=P239_Alv_MDM_h6[i,]/colSums(P239_Alv_MDM_h6)*1e6,
    "alv_gfpneg2"=M239_N239_Alv_h3[i,]/colSums(M239_N239_Alv_h3)*1e6,
    "alv_gfpneg3"=M239_N239_Alv_h6[i,]/colSums(M239_N239_Alv_h6)*1e6)

  lgfp <- lapply(gfp,function(x) { log10(x+1) } )
  boxplot(lgfp,col="white",cex=0,las=2,border="darkgray",ylab="log10+1 RPM", main=rownames(mdm_mock1)[i])
  boxplot(lgfp,col="white",cex=0,las=2,border="darkgray",ylab="log10+1 RPM", main=rownames(mdm_mock1)[i])
  beeswarm(lgfp,add=TRUE,pch=1,cex=0.1)
})

par(mar=c(5.1, 4.1, 4.1, 2.1))

#pdf("latent_threshold.pdf",width=7,height=9)

# mdm1 mock vs active
m1a <- log10( colSums(mdm_active1[1:21,]) / colSums(mdm_active1) *1e6 )
m1m <- log10( colSums(mdm_mock1[1:21,]) / colSums(mdm_mock1) *1e6 )
m1n <- log10(colSums(gex1_h4_d0[1:21,]) / colSums(gex1_h4_d0) *1e6)
myrange <- seq(1,6,0.1)
m1res <- sapply( myrange , function(j) {
  ( length(which(m1m < j))/length(m1m) + length(which(m1a >= j))/length(m1a) ) / 2  }
)
names(m1res) <- myrange
MAX=as.numeric(names(m1res[order(-m1res)][1]))
MAXVAL=signif(m1res[order(-m1res)][1],3)
par(mfrow=c(4,1))
hist( m1m , main="MDM1 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( m1a , main="MDM1 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
plot(myrange,m1res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( m1n , main="MDM1 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(m1n>=MAX)), collapse=" "),cex=0.7)
mdm_latent1 <- gex1_h4_d0[,which(m1n>=MAX)] #MDMP236 latent
mdm_bystander1 <- gex1_h4_d0[,which(m1n<MAX)] #MDMP236 bystander

# mdm2 mock vs active
m2a <- log10( colSums(mdm_active2[1:21,]) / colSums(mdm_active2) *1e6 )
m2m <- log10( colSums(mdm_mock2[1:21,]) / colSums(mdm_mock2) *1e6 )
m2n <- log10( colSums(gex1_h4_d1[1:21,]) / colSums(gex1_h4_d1) *1e6)
myrange <- seq(1,6,0.1)
m2res <- sapply( myrange , function(j) {
  ( length(which(m2m < j))/length(m2m) + length(which(m2a >= j))/length(m2a) ) / 2  }
)
names(m2res) <- myrange
MAX=as.numeric(names(m2res[order(-m2res)][1]))
MAXVAL=signif(m2res[order(-m2res)][1],3)
par(mfrow=c(4,1))
hist( m2m , main="MDM2 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( m2a , main="MDM2 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
plot(myrange,m2res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( m2n , main="MDM2 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(m2n>=MAX)), collapse=" "),cex=0.7)
mdm_latent2 <- gex1_h4_d1[,which(m2n>=MAX)] #MDMV236 latent
mdm_bystander2 <- gex1_h4_d1[,which(m2n<MAX)] #MDMV236 bystander

# mdm3 mock vs active
m3a <- log10( colSums(mdm_active3[1:21,]) / colSums(mdm_active3) *1e6 )
m3m <- log10( colSums(mdm_mock3[1:21,]) / colSums(mdm_mock3) *1e6 )
m3n <- log10( colSums(gex2_h4[1:21,]) / colSums(gex2_h4) *1e6)
myrange <- seq(1,6,0.1)
m3res <- sapply( myrange , function(j) {
  ( length(which(m3m < j))/length(m3m) + length(which(m3a >= j))/length(m3a) ) / 2  }
)
names(m3res) <- myrange
MAX=as.numeric(names(m3res[order(-m3res)][1]))
MAXVAL=signif(m3res[order(-m3res)][1],3)
par(mfrow=c(4,1))
hist( m3m , main="MDM3 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( m3a , main="MDM3 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
plot(myrange,m3res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( m3n , main="MDM3 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(m3n>=MAX)), collapse=" "),cex=0.7)
mdm_latent3 <- gex2_h4[,which(m3n>=MAX)] #MDMO236 latent
mdm_bystander3 <- gex2_h4[,which(m3n<MAX)] #MDMO236 bystander

# mdm4 mock vs active
m4a <- log10( colSums(mdm_active4[1:21,]) / colSums(mdm_active4) *1e6 )
m4m <- log10( colSums(mdm_mock4[1:21,]) / colSums(mdm_mock4) *1e6 )
m4n <- log10( colSums(P239_Alv_MDM_h3[1:21,]) / colSums(P239_Alv_MDM_h3) *1e6)
myrange <- seq(1,6,0.1)
m4res <- sapply( myrange , function(j) {
  ( length(which(m4m < j))/length(m4m) + length(which(m4a >= j))/length(m4a) ) / 2  }
)
names(m4res) <- myrange
MAX=as.numeric(names(m4res[order(-m4res)][1]))
MAXVAL=signif(m4res[order(-m4res)][1],3)
par(mfrow=c(4,1))
hist( m4m , main="MDM4 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( m4a , main="MDM4 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
plot(myrange,m4res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( m4n , main="MDM4 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(m4n>=MAX)), collapse=" "),cex=0.7)
mdm_latent4 <- P239_Alv_MDM_h3[,which(m4n>=MAX)] #MDMP239 latent
mdm_bystander4 <- P239_Alv_MDM_h3[,which(m4n<MAX)] #MDMP239 bystander

# alv1 mock vs active
a1a <- log10( colSums(alv_active1[1:21,]) / colSums(alv_active1) *1e6 )
a1m <- log10( colSums(alv_mock1[1:21,]) / colSums(alv_mock1) *1e6 )
a1n <- log10( colSums(P239_Alv_MDM_h6[1:21,]) / colSums(P239_Alv_MDM_h6) *1e6)
myrange <- seq(1,6,0.1)
a1res <- sapply( myrange , function(j) {
  ( length(which(a1m < j))/length(a1m) + length(which(a1a >= j))/length(a1a) ) / 2  }
)
names(a1res) <- myrange
MAX=as.numeric(names(a1res[order(-a1res)][1]))
MAXVAL=signif(a1res[order(-a1res)][1],3)
par(mfrow=c(4,1))
hist( a1m , main="Alv1 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( a1a , main="Alv1 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
plot(myrange,a1res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( a1n , main="Alv1 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(a1n>=MAX)), collapse=" "),cex=0.7)
alv_latent1 <- P239_Alv_MDM_h6[,which(a1n>=MAX)] #AlvP239 latent
alv_bystander1 <- P239_Alv_MDM_h6[,which(a1n<MAX)] #AlvP239 bystander

# alv2 mock vs active
a2a <- log10( colSums(alv_active2[1:21,]) / colSums(alv_active2) *1e6 )
a2m <- log10( colSums(alv_mock2[1:21,]) / colSums(alv_mock2) *1e6 )
a2n <- log10( colSums(M239_N239_Alv_h3[1:21,]) / colSums(M239_N239_Alv_h3) *1e6)
myrange <- seq(1,6,0.1)
a2res <- sapply( myrange , function(j) {
  ( length(which(a2m < j))/length(a2m) + length(which(a2a >= j))/length(a2a) ) / 2  }
)
names(a2res) <- myrange
MAX=as.numeric(names(a2res[order(-a2res)][1]))
MAXVAL=signif(a2res[order(-a2res)][1],3)
par(mfrow=c(4,1))
hist( a2m , main="Alv2 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( a2a , main="Alv2 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
plot(myrange,a2res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( a2n , main="Alv2 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(a2n>=MAX)), collapse=" "),cex=0.7)
alv_latent2 <- M239_N239_Alv_h3[,which(a2n>=MAX)] #AlvM239 latent
alv_bystander2 <- M239_N239_Alv_h3[,which(a2n<MAX)] #AlvM239 bystander

# alv3 mock vs active
a3a <- log10( colSums(alv_active3[1:21,]) / colSums(alv_active3) *1e6 )
a3m <- log10( colSums(alv_mock3[1:21,]) / colSums(alv_mock3) *1e6 )
a3n <- log10( colSums(M239_N239_Alv_h6[1:21,]) / colSums(M239_N239_Alv_h6) *1e6)
myrange <- seq(1,6,0.1)
a3res <- sapply( myrange , function(j) {
  ( length(which(a3m < j))/length(a3m) + length(which(a3a >= j))/length(a3a) ) / 2  }
)
names(a3res) <- myrange
MAX=as.numeric(names(a3res[order(-a3res)][1]))
MAXVAL=signif(a3res[order(-a3res)][1],3)
par(mfrow=c(4,1))
hist( a3m , main="Alv3 mock" , xlab="log10 HIV RPM", xlim=c(1,6), breaks=25)
hist( a3a , main="Alv3 active", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
plot(myrange,a3res,type="b",main="Best partition",xlab="log10 HIV RPM",bty="n")
abline(v=MAX,col="red")
mtext(paste("Max=",MAXVAL,"@",MAX),cex=0.7)
hist( a3n , main="Alv3 GFP-", xlab="log10 HIV RPM", xlim=c(1,6), breaks=35)
abline(v=MAX,col="red")
mtext(paste(as.character(table(a3n>=MAX)), collapse=" "),cex=0.7)
alv_latent3 <- M239_N239_Alv_h6[,which(a3n>=MAX)] #AlvN239 latent
alv_bystander3 <- M239_N239_Alv_h6[,which(a3n<MAX)] #AlvN239 bystander

#dev.off()

react6 <- N239Alv_react_counts #AlvN239

colnames(mdm_mock1) <- paste("mdm_mock1|",colnames(mdm_mock1),sep="")
colnames(mdm_mock2) <- paste("mdm_mock2|",colnames(mdm_mock2),sep="")
colnames(mdm_mock3) <- paste("mdm_mock3|",colnames(mdm_mock3),sep="")
colnames(mdm_mock4) <- paste("mdm_mock4|",colnames(mdm_mock4),sep="")
colnames(alv_mock1) <- paste("alv_mock1|",colnames(alv_mock1),sep="")
colnames(alv_mock2) <- paste("alv_mock2|",colnames(alv_mock2),sep="")
colnames(alv_mock3) <- paste("alv_mock3|",colnames(alv_mock3),sep="")

colnames(mdm_active1) <- paste("mdm_active1|",colnames(mdm_active1),sep="")
colnames(mdm_active2) <- paste("mdm_active2|",colnames(mdm_active2),sep="")
colnames(mdm_active3) <- paste("mdm_active3|",colnames(mdm_active3),sep="")
colnames(mdm_active4) <- paste("mdm_active4|",colnames(mdm_active4),sep="")

colnames(alv_active1) <- paste("alv_active1|",colnames(alv_active1),sep="")
colnames(alv_active2) <- paste("alv_active2|",colnames(alv_active2),sep="")
colnames(alv_active3) <- paste("alv_active3|",colnames(alv_active3),sep="")

colnames(mdm_bystander1) <- paste("mdm_bystander1|",colnames(mdm_bystander1),sep="")
colnames(mdm_bystander2) <- paste("mdm_bystander2|",colnames(mdm_bystander2),sep="")
colnames(mdm_bystander3) <- paste("mdm_bystander3|",colnames(mdm_bystander3),sep="")
colnames(mdm_bystander4) <- paste("mdm_bystander4|",colnames(mdm_bystander4),sep="")

colnames(alv_bystander1) <- paste("alv_bystander1|",colnames(alv_bystander1),sep="")
colnames(alv_bystander2) <- paste("alv_bystander2|",colnames(alv_bystander2),sep="")
colnames(alv_bystander3) <- paste("alv_bystander3|",colnames(alv_bystander3),sep="")

colnames(mdm_latent1) <- paste("mdm_latent1|",colnames(mdm_latent1),sep="")
colnames(mdm_latent2) <- paste("mdm_latent2|",colnames(mdm_latent2),sep="")
colnames(mdm_latent3) <- paste("mdm_latent3|",colnames(mdm_latent3),sep="")
colnames(mdm_latent4) <- paste("mdm_latent4|",colnames(mdm_latent4),sep="")

colnames(alv_latent1) <- paste("alv_latent1|",colnames(alv_latent1),sep="")
colnames(alv_latent2) <- paste("alv_latent2|",colnames(alv_latent2),sep="")
colnames(alv_latent3) <- paste("alv_latent3|",colnames(alv_latent3),sep="")

colnames(react6) <- paste("react6|",colnames(react6),sep="")

mylist <- list(mdm_mock1,mdm_mock2,mdm_mock3,mdm_mock4,alv_mock1,alv_mock2,alv_mock3,
  mdm_active1,mdm_active2,mdm_active3,mdm_active4,alv_active1,alv_active2,alv_active3,
  mdm_latent1,mdm_latent2,mdm_latent3,mdm_latent4,alv_latent1,alv_latent2,alv_latent3,
  mdm_bystander1,mdm_bystander2,mdm_bystander3,mdm_bystander4,alv_bystander1,
  alv_bystander2,alv_bystander3,react6)

names(mylist) <- c("mdm_mock1","mdm_mock2","mdm_mock3","mdm_mock4","alv_mock1","alv_mock2",
  "alv_mock3", "mdm_active1","mdm_active2","mdm_active3","mdm_active4","alv_active1",
  "alv_active2","alv_active3", "mdm_latent1","mdm_latent2","mdm_latent3","mdm_latent4",
  "alv_latent1","mdm_latent2","alv_latent3", "mdm_bystander1","mdm_bystander2",
  "mdm_bystander3","mdm_bystander4","alv_bystander1","alv_bystander2","alv_bystander3",
  "react6")

comb <- do.call(cbind,mylist)

rev(sapply(mylist,ncol))
barplot(rev(sapply(mylist,ncol)),horiz=TRUE,las=1,xlab="no. cells")

rev(sapply(mylist,function(mx) {sum(colSums(mx))}))

par(mar = c(5.1, 10.1, 4.1, 2.1))

barplot(rev(sapply(mylist,function(mx) {sum(colSums(mx))})),
  horiz=TRUE,las=1,xlab="no. counts")

res <- rev(sapply(mylist,function(mx) {
  mean(colSums(mx[1:2,])) /  mean(colSums(mx[3:nrow(mx),])) *100
}))
res
barplot(rev(res),horiz=TRUE,las=1,xlab="% HIV reads")

par(mar = c(5.1, 4.1, 4.1, 2.1))

```

## Analysis by chromosome

```{r,chranalysis1}

chrmx <- sapply(mylist,function(mx) {
  rsum <- as.data.frame(rowSums(mx))
  m <- merge(gene_chr,rsum,by.x="V2",by.y=0)
  m$V2=NULL
  ag <- aggregate(. ~ V1 , m , sum)
  out <- ag[,2]
  names(out) <- ag[,1]
  return(out)
})

chryx <- chrmx["chrY",] / chrmx["chrX",] * 100
chryx
par(mar = c(5.1, 10.1, 4.1, 2.1))
barplot(rev(chryx),horiz=TRUE,las=1,xlab="Ratio ChrY to ChrX x100")

```

These checks look consistent, except for the react6 which appears as more male but that is not consistent with
mock6, active6 and latent6.

```{r,chranalysis2}

chrmt <- chrmx["chrM",] / colSums(chrmx[1:22,]) *100
chrmt
barplot(rev(chrmt),horiz=TRUE,las=1,xlab="% chrM reads")

```

Let's take a look at mtDNA transcripts at the level of individual cells.

Remove cells with >20% mito reads.

```{r,chrmtbycell}

mt <- lapply(mylist, function(x) {
  xmt <- x[grep("MT-",rownames(x)),]
  sapply(1:ncol(x), function(j) { sum(xmt[,j])/sum(x[,j]) } )
} )

vioplot(mt)

boxplot(mt,horizontal=TRUE,cex=0,col="white",las=1,
 main="mtDNA gene proportion")

barplot(sapply(mt, function(x) { length(which(x>0.2))/length(x) } ),horiz=TRUE,las=1, main="Proportion of cells with >15% mito reads")

fmt <- lapply(mylist, function(x) {
  xmt <- x[grep("MT-",rownames(x)),]
  v <- sapply(1:ncol(x), function(j) { sum(xmt[,j])/sum(x[,j]) } )
  f <-  x[,which(v<0.2)]
  return(f)
} )

lapply(fmt,dim)

mt2 <- lapply(fmt, function(x) {
  xmt <- x[grep("MT-",rownames(x)),]
  sapply(1:ncol(x), function(j) { sum(xmt[,j])/sum(x[,j]) } )
} )

vioplot(mt2)

boxplot(mt2,horizontal=TRUE,cex=0,col="white",las=1,
 main="mtDNA gene proportion after filtering")

```

## Save data object

```{r,save}

saveRDS(fmt,"macrophage_counts.rds")

```

## Session information

For reproducibility.

```{r,session}

save.image("macrophage_dataprep.Rdata")


sessionInfo()

```
